### PR TITLE
Feature/price override

### PR DESCRIPTION
**Pull request by sharmarkei**

### CODE REVIEW

File: route/price-router.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+const Router = require('express').Router;
+const createError = require('http-errors');
+const debug = require('debug')('parkify:price-router');
+const jsonParser = require('body-parser').json();
+const bearerAuth = require('../lib/bearer-auth-middleware.js');
+const Price = require('../model/price.js');
+
+const priceRouter = module.exports = Router();
+
+priceRouter.post('/api/price', bearerAuth, jsonParser, function (request, response, next) {
```
Comment (ohjonah) [inline route/price-router.js]: can you get rid of the space between function and (

File: route/price-router.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    .then(price => {
+      response.json(price);
+    })
+    .catch(error => {
+      error = createError(400, error.message);
+      next(error);
+    });
+});
+
+priceRouter.get('/api/price/:id', bearerAuth, function (request, response, next) {
```
Comment (ohjonah) [inline route/price-router.js]: can you get rid of the space between function and (

File: route/price-router.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+  Price.findById(request.params.id)
+    .then(price => {
+      if (!price) return next(createError(404, 'No price found'));
+      response.json(price);
+    })
+    .catch(err => next(createError(404, err.message)));
+});
+
+priceRouter.put('/api/price/:id', bearerAuth, jsonParser, function (request, response, next) {
```
Comment (ohjonah) [inline route/price-router.js]: can you get rid of the space between function and (

File: route/price-router.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  debug('PUT: /api/price/:id');
+
+  Price.findByIdAndUpdate(request.params.id, request.body, { new: true })
+    .then(price => response.json(price))
+    .catch(err => next(createError(404, err.message)));
+});
+
+priceRouter.delete('/api/price/:id', bearerAuth, function (request, response, next) {
```
Comment (ohjonah) [inline route/price-router.js]: can you get rid of the space between function and (

File: route/price-router.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+const createError = require('http-errors');
+const debug = require('debug')('parkify:price-router');
+const jsonParser = require('body-parser').json();
+const bearerAuth = require('../lib/bearer-auth-middleware.js');
+const Price = require('../model/price.js');
+
+const priceRouter = module.exports = Router();
+
+priceRouter.post('/api/price', bearerAuth, jsonParser, function (request, response, next) {
+  debug('POST: /api/price');
```
Comment (ohjonah) [inline route/price-router.js]: Needs a more specific route

File: route/price-router.js (hunk 3 above)
Comment (ohjonah) [inline route/price-router.js]: Needs a specific route to lot for it to be overridden.

File: route/price-router.js (hunk 4 above)
Comment (ohjonah) [inline route/price-router.js]: Needs a specific route to lot for it to be overridden.

**Review by nickjaz — APPROVED** [review]

**Pull request merged**